workflow-templates/trivy.yml

```diff
@@ -0,0 +1,46 @@
+name: trivy
+
+on:
+  pull_request:
+    branches: [ "main", "develop" ]
+  schedule:
+    - cron: '0 0 1 * *'
+  push:
+    branches: [ "main", "develop" ]
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    permissions:
+      contents: read 
+      security-events: write
+
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Build an image from Dockerfile
+        run: |
+          docker build -t docker.io/kbase/trivy:scan .
+
+      - name: Run Trivy vulnerability scanner
+        uses: aquasecurity/trivy-action@master
+        with:
+          image-ref: "kbase/trivy:scan"
+          format: "sarif"
+          output: "trivy-results.sarif"
+          timeout: "20m"
+          ignore-unfixed: true
+          severity: 'HIGH,CRITICAL'
+          limit-severities-for-sarif: true
+          exit-code: '1' # Fail Step if CVE of specified severity is detected
+
+      - name: Upload Trivy scan results to GitHub Security tab
+        if: always()
+        uses: github/codeql-action/upload-sarif@v3
+        with:
+          sarif_file: 'trivy-results.sarif'
```

Comment:
doesn't this need to upload to the branch / commit?

Reply:
Ok, so `on pull_request`` -> uploads to PR, `on push` -> uploads to branch?

Does that mean the old trivy scan in pr_build is obsoleted by this?

Reply:
* Yes
* No, it doesn't run on push

Reply:
I'm confused by point #2 - why would that mean the old trivy scan isn't obsolete? Unless I'm confused the scan in this PR does everything the old scan does other than check for those specific CVEs you mention later

Reply:
No, it doesn't error on those 3 specific CVEs.

Reply:
Original Trivy Scan Error Conditions
* Any of CVES found in list of 3 log4j cves

New Trivy Scan Error Conditions
* Errors on log4j cves, and ANY other found within severity level

So they are doing different things. If you